### PR TITLE
feat: add `userAgent` to `RESTOptions`

### PR DESCRIPTION
**Pull request by tjrgg**
I've added `userAgent` to RESTOptions to allow for a custom User-Agent. By default, the option uses the one that was defined, but I've defined it in the RESTOptions and removed the export, since it was no longer necessary.

There probably isn't a lot of uses for this, but I didn't want to see this ability taken away by using this package in a package not associated with Dirigeants or Project Blue.

### CODE REVIEW

**Comment by PyroTechniac** [comment]
The User Agent, as per Discord API Documentation, is meant to give discord information about the client library being used, and should not be changed by end users.

**Comment by tjrgg** [comment]
My use case for having this option would be to use this package as part of another client library. Since this is built as a separate package, it is possible that developers like myself may choose to use this package as part of other client libraries and/or frameworks, not just in association with Project Blue. As such, I don't think it makes sense to identify those other libraries to Discord as this project. Given this possibility, I don't think it makes sense to restrict the ability for changing the User-Agent.

File: src/lib/RESTManager.ts [hunk 1]
```diff
@@ -128,7 +129,7 @@ export class RESTManager {
 
 		// Assign the basic request headers
 		const headers: Headers = {
-			'User-Agent': UserAgent,
+			'User-Agent': this.options.userAgent,
```
Comment (cfanoulis) [inline src/lib/RESTManager.ts]: as discussed in the diri server pls make this conform to the discord spec thanks

**Comment by tjrgg** [comment]
Closing in favor of an option to append to the userAgent instead.

**Pull request closed without merging**